crates/data-generator/src/blocks/json.rs

```diff
@@ -0,0 +1,153 @@
+use std::collections::BTreeMap;
+
+use indexmap::IndexMap;
+use serde::Deserialize;
+
+use super::model::{Block, Id, PropertyVariant, PropertyVariants, State, States};
+use crate::name::Name;
+
+#[derive(Debug, Deserialize)]
+struct SerdeId(u32);
+
+#[derive(Debug, Deserialize)]
+struct SerdeBlock {
+    #[serde(default)]
+    properties: BTreeMap<String, Vec<String>>,
+    states: Vec<SerdeState>,
+}
+
+#[derive(Debug, Deserialize)]
+struct SerdeState {
+    id: SerdeId,
+    #[serde(default)]
+    default: bool,
+    #[serde(default)]
+    properties: BTreeMap<String, String>,
+}
+
+#[derive(Debug, thiserror::Error)]
+pub enum ParseError {
+    #[error("malformed JSON data")]
+    Json(#[from] serde_json::Error),
+}
+
+pub(super) fn parse_block_registry(json_data: String) -> Result<IndexMap<Name, Block>, ParseError> {
```

Comment:
```suggestion
pub fn parse_block_registry(json_data: String) -> Result<IndexMap<Name, Block>, ParseError> {
```
Unnecessary encapsulation